src/components/Convert/routes/hlpBalancer.ts

```diff
@@ -38,8 +42,28 @@ const hlpBalancerWeight = async ({ fromToken, toToken, network }: WeightInput):
   return BALANCER_TO_HLP;
 };
 
+const convertEthToWeth = (path: BalancerPath, network: Network): BalancerPath => {
+  const weth = mustExist(
+    HandleTokenManagerInstance.getHlpWrappedNativeToken(network),
+    `WETH on ${network}`
+  );
+
+  // treat eth as weth
+  if (path.tokenIn.extensions?.isNative) {
+    path.tokenIn = weth;
+  }
+
+  if (path.tokenOut.extensions?.isNative) {
+    path.tokenOut = weth;
+  }
+
+  return path;
+};
+
 const getHlpToBalancerAmountAndFees = async (input: ConvertQuoteRouteArgs) => {
-  const path = getHlpToBalancerPath(input.fromToken, input.toToken);
+  let path = getHlpToBalancerPath(input.fromToken, input.toToken);
+
+  path = convertEthToWeth(path, input.network);

```

Comment:
```suggestion
  const path = convertEthToWeth(
    getHlpToBalancerPath(input.fromToken, input.toToken),
    input.network
  );
```
Can you do the same for the other file too please?